### PR TITLE
Fixes no sponsored benefit issue

### PR DESCRIPTION
**Pull request by utkarsh7989**
RM-97080

### CODE REVIEW

**Review by vineelsonnathi — DISMISSED** [review]
LGTM

**Review by markgoho — DISMISSED** [review]
@utkarsh7989 if the ticket represents a bug or you are clearly fixing something that's broken - it needs to have a spec change associated with it

**Review by vineelsonnathi — DISMISSED** [review]
LGTM. 
Specs written.

File: components/benefit_sponsors/spec/views/benefit_sponsors/benefit_applications/edit.html.slim_spec.rb [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    allow(view).to receive(:product_rates_available?).and_return(true)
+    allow(view).to receive(:benefit_sponsorship_benefit_application_path).and_return('/')
+    # view.should_receive(:render).with(hash_including(:partial => "form")).and_return("<span id=\"rendered-form\"/>")
+    allow(view).to receive(:render).and_return("<span id=\"rendered-form\"/>")
+    sign_in user
+    render template: "benefit_sponsors/benefit_applications/benefit_applications/edit"
+  end
+
+  it 'should not show Reference Plans if health sponsored benefits does not present' do
+    expect(rendered).to_not match(/Reference Plans/)
```
Comment (markgoho) [inline components/benefit_sponsors/spec/views/benefit_sponsors/benefit_applications/edit.html.slim_spec.rb:40]: please do not use hard-coded english text to assert on, please use an element, or a css class/attribute

**Review by markgoho — CHANGES_REQUESTED** [review]

**Review by vineelsonnathi — APPROVED** [review]
LGTM.

**Review by markgoho — APPROVED** [review]

**Pull request merged**